_episodes/06-raster-intro.md

```diff
@@ -0,0 +1,433 @@
+print(raster_ams_b9.mean())
+print(raster_ams_b9.std())
+~~~
+{: .language-python}
+~~~
+<xarray.DataArray ()>
+array(8., dtype=float32)
+Coordinates:
+    spatial_ref  int64 0
+<xarray.DataArray ()>
+array(15497., dtype=float32)
+Coordinates:
+    spatial_ref  int64 0
+<xarray.DataArray ()>
+array(2477.405, dtype=float32)
+Coordinates:
+    spatial_ref  int64 0
+<xarray.DataArray ()>
+array(2061.9539, dtype=float32)
+Coordinates:
+    spatial_ref  int64 0
+~~~
+{: .output}
+
+And if we plot the image, the `nodata` pixels are not shown because they are not 0 anymore:
+
+![Raster plot with rioxarray using the robust setting no data](../fig/E06-03-overview-plot-B09-robust-with-nan.png)
+
+One should notice that there is a side effect of using `nan` instead of `0` to represent the missing data: the data type of the `DataArray` was changed from integers to float. This need to be taken into consideration when the data type matters in your application.
+
+## Raster Bands
+So far we looked into a single band raster, i.e. the `B09` band of a Sentinel-2 scene. However, to get an overview of the scene, one may also want to visualize the true-color thumbnail of the region. This is provided as a multi-band raster -- a raster dataset that contains more than one band. 
+
+![Multi-band raster image](../images/dc-spatial-raster/single_multi_raster.png)
+
+The `overview` asset in the Sentinel-2 scene is a multiband asset. Similar to `B09`, we can load it by:
+~~~
+raster_ams_overview = rioxarray.open_rasterio(items[0].assets['overview'].href)
+~~~
+{: .language-python}
+~~~
+<xarray.DataArray (band: 3, y: 343, x: 343)>
+[352947 values with dtype=uint8]
+Coordinates:
+  * band         (band) int64 1 2 3
+  * x            (x) float64 6.002e+05 6.005e+05 ... 7.093e+05 7.096e+05
+  * y            (y) float64 5.9e+06 5.9e+06 5.899e+06 ... 5.791e+06 5.79e+06
+    spatial_ref  int64 0
+Attributes:
+    _FillValue:    0.0
+    scale_factor:  1.0
+    add_offset:    0.0
+~~~
+{: .output}
+
+The band number comes first when GeoTiffs are read with the `.open_rasterio()` function. As we can see in the `xarray.DataArray` object, the shape is `(band: 3, y: 343, x: 343)` now. There are three bands in the `xarray.DataArray` now. It's always a good idea to examine the shape of the raster array you are working with and make sure it's what you expect. Many functions, especially the ones that plot images, expect a raster array to have a particular shape. One can also check the shape using the `.shape` attribute:
+~~~
+raster_ams_overview.shape
+~~~
+{: .language-python}
+~~~
+(3, 343, 343)
+~~~
+{: .output}
+
+One can visualize the multi-band data with the `DataArray.plot.imshow()` function:
+~~~
+raster_ams_overview.plot.imshow()
+~~~
+{: .language-python}
+
+![Amsterdam true color overview](../fig/E06-04-overview-plot-true-color.png)
+
+Note that the `DataArray.plot.imshow()` function make pre-assumptions on the RGB channels. It does not work directly on the multi-band data.
```

Comment:
```suggestion
Note that the `DataArray.plot.imshow()` function makes assumptions about the shape of the input DataArray, that since it has three channels, the correct colormap for these channels is RGB. It does not work directly on image arrays with more than 3 channels. However, the `cmap` argument can be used to create false color composites.
```

Reply:
should we show a false color composite example? could be cool to show the power of imshow and that rioxarray isn't purely limited to RGB data, which I could see folks who are new assuming based on the original description.

Reply:
Hi Ryan, indeed this is more interesting! The general motivation of plotting the `overview` is because it is simple. 

To visualize a false image, we need to:

- Retrieve B02 (Blue), B03(Green) and B08(NIR) respectively
- Downsample them to a plottable size with `reproject function`
- Normalize the value to [0,255)
- Concrate the bands with with `xarray`

IMO this is a bit overcomplicated in the raster intro. I can a briefly mention that one could also plot a false value image by XYZ steps. Do you think this is sufficient?

Reply:
I think your edit and explanation of false color is correct. The cmap arg wouldn't actually change, my mistake. This looks good for now, another episode can contain this example.